packages/connect-tokens/src/thegraph/connector.ts

```diff
@@ -29,21 +36,28 @@ export default class TokensConnectorTheGraph implements ITokensConnector {
   }
 
   static async create(
-    subgraphUrl: string,
-    appAddress: Address,
-    verbose: boolean = false
+    config: TokensConnectorTheGraphConfig
   ): Promise<TokensConnectorTheGraph> {
-    if (!subgraphUrl) {
+    if (!config.subgraphUrl) {
       throw new Error(
         'TokensConnectorTheGraph requires subgraphUrl to be passed.'
       )
     }
 
-    const gql = new GraphQLWrapper(subgraphUrl, verbose)
+    if (!config.appAddress) {
+      throw new Error(
+        'TokensConnectorTheGraph requires appAddress to be passed.'
+      )
+    }
+
+    const gql = new GraphQLWrapper(config.subgraphUrl, {
+      pollInterval: config.pollInterval,
+      verbose: config.verbose,
+    })
 
     const token = await gql.performQueryWithParser<Token>(
       queries.TOKEN('query'),
-      { tokenManagerAddress: appAddress },
+      { tokenManagerAddress: config.appAddress },
```

Comment:
Using `create()` is needed in the tokens app connector, because it’s the only one that need to instantiate asynchronously. What would you think of moving all the connectors to an async `create()`, even if they don’t need it (yet)?